src/Nethermind/Nethermind.Blockchain.Test/TransactionSelectorTests.cs

```diff
@@ -139,45 +313,113 @@ void SetAccountStates(IEnumerable<Address> missingAddresses)
             }
 
             ITxPool transactionPool = Substitute.For<ITxPool>();
-            IComparer<Transaction> comparer = CompareTxByNonce.Instance.ThenBy(TxPool.TxPool.DefaultComparer);
-            transactionPool.GetPendingTransactionsBySender().Returns(
-                testCase.Transactions
-                    .Where(t => t?.SenderAddress != null)
-                    .GroupBy(t => t.SenderAddress)
-                    .ToDictionary(
-                    g => g.Key, 
-                    g => g.OrderBy(t => t, comparer).ToArray()));
-            
+            IBlockTree blockTree = Substitute.For<IBlockTree>();
+            Block block = Build.A.Block.WithNumber(0).TestObject;
+            blockTree.Head.Returns(block);
+            IReleaseSpec spec = new ReleaseSpec()
+            {
+                IsEip1559Enabled = testCase.Eip1559Enabled
+            };
+            specProvider.GetSpec(Arg.Any<long>()).Returns(spec);
+            TransactionComparerProvider transactionComparerProvider =
+                new TransactionComparerProvider(specProvider, blockTree);
+            IBlockPreparationContextService blockPreparationContextService = new BlockPreparationContextService(LimboLogs.Instance);
+            blockPreparationContextService.SetContext(testCase.BaseFee, 0);
+            IComparer<Transaction> defaultComparer = transactionComparerProvider.GetDefaultComparer();
+            IComparer<Transaction> comparer = CompareTxByNonce.Instance.ThenBy(defaultComparer);
+            Dictionary<Address?, Transaction[]> transactions = testCase.Transactions
```

Comment:
this test seems overengineered which means that when it breaks the fix may be very time consuming and also it may be hard to understand what the test is testing

Reply:
Hmmm... I think it is a good integration test. It checks if TxPoolTxSource returns expected results. Perhaps, we should add also unit tests for example for BaseFeeTxFilter and other filtering classes that create a full filtering pipeline. We can discuss it on any occasion :)

What is more, we have tests description like: "EIP1559 transactions: All transactions selected - BaseFee lower than gas price", "None transactions selected due to transaction gas price and limit".